File: weekend-core/src/main/java/com/github/aly8246/core/template/BaseTemplate.java
package com.github.aly8246.core.template;

import com.github.aly8246.core.annotation.Command;
import org.springframework.util.StringUtils;

public class BaseTemplate implements Template {

@Override
public String completeCommand(Command command) {
	String baseCommand = String.join("", command.value());
	if (StringUtils.isEmpty(baseCommand)) throw new RuntimeException("BaseCommand不能为空");
	return baseCommand;
}
//TODO 先将模板替换成具体参数值
// args command
// 完整命令

}

File: weekend-core/src/main/java/com/github/aly8246/core/template/BaseTemplate.kt
package com.github.aly8246.core.template

import com.github.aly8246.core.util.StringUtils

abstract class BaseTemplate : Template {

    override fun completeCommand(baseCommand: String, param: MutableMap<String, Any?>): String {
        when {
            StringUtils.isEmpty(baseCommand) -> throw RuntimeException("BaseCommand不能为空")
        }
        val replaceParam = replaceParam(baseCommand, param)
        syntaxCheck(replaceParam)
        return replaceParam
    }

    abstract fun replaceParam(sourceCommand: String, param: MutableMap<String, Any?>): String

    abstract fun syntaxCheck(command: String)

}
